sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessPipelineRunner.java

```diff
@@ -189,6 +208,14 @@ public InProcessPipelineOptions getPipelineOptions() {
     return options;
   }
 
+  Supplier<Clock> getClockSupplier() {
+    return clockSupplier;
+  }
+
+  void setClockSupplier(Supplier<Clock> supplier) {
```

Comment:
The runner will obtain an instance of the Clock during the call to `run()` and pass it to the `EvaluationContext`; so each is independent.